test/functional/mempool_accept.py

```diff
@@ -327,6 +328,54 @@ def run_test(self):
             rawtxs=[tx.serialize().hex()],
         )
 
+        # OP_RETURN followed by non-push
+        tx = tx_from_hex(raw_tx_reference)
+        tx.vout[0].scriptPubKey = CScript([OP_RETURN, OP_HASH160])
+        self.check_mempool_result(
+            result_expected=[{'txid': tx.rehash(), 'allowed': False, 'reject-reason': 'scriptpubkey'}],
+            rawtxs=[tx.serialize().hex()],
+        )
+
+        # Multiple OP_RETURN and more than 83 bytes, even if over MAX_SCRIPT_ELEMENT_SIZE
+        # are standard since v30
+        tx = tx_from_hex(raw_tx_reference)
+        tx.vout.append(CTxOut(0, CScript([OP_RETURN, b'\xff'])))
+        tx.vout.append(CTxOut(0, CScript([OP_RETURN, b'\xff' * 50000])))
```

Comment:
I was a bit surprised that we accept this output script. I thought that the OP_RETURN opcode is to be followed by a push with the length of the data payload, but it looks like you are just writing raw data right after OP_RETURN. Could you clarify why no push is necessary for this to be accepted?

Reply:
It's never executed and therefore does not have to be valid Script.

Reply:
@murchandamus: The test framework's CScript class automatically generates data pushes for byte-strings that are passed in the list, e.g.:
```
$ python3
Python 3.12.9 (main, Apr  9 2025, 12:59:26) [Clang 16.0.6 ] on openbsd7
Type "help", "copyright", "credits" or "license" for more information.
>>> from test_framework.script import CScript, OP_RETURN
>>> s = CScript([OP_RETURN, b'meh'])
>>> s.hex()
'6a036d6568'
```

> It's never executed and therefore does not have to be valid Script.

True from a consensus perspective, but an OP_RETURN output without data pushes wouldn't be treated as standard and hence rejected by the mempool:

https://github.com/bitcoin/bitcoin/blob/ae024137bda9fe189f4e7ccf26dbaffd44cbbeb6/src/script/solver.cpp#L182-L187